python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +615,1002 @@ def build_quantum_graph(
+        # outside of the butler.  This task has been given (temporary)
+        # Special Dispensation because of the nature of HiPS outputs until
+        # a more controlled solution can be found.
+
+        # Make the initial list of UNIQ pixels
+        uniq = 4*(4**max_order) + pixels
+
+        # Make a healsparse map which provides easy degrade/comparisons.
+        hspmap = hsp.HealSparseMap.make_empty(2**min_uniq_order, 2**max_order, dtype=np.float32)
+        hspmap[pixels] = 1.0
+
+        # Loop over orders, degrade each time, and look for pixels with full coverage.
+        for uniq_order in range(max_order - 1, min_uniq_order - 1, -1):
+            hspmap = hspmap.degrade(2**uniq_order, reduction="sum")
+            pix_shift = np.right_shift(pixels, 2*(max_order - uniq_order))
+            # Check if any of the pixels at uniq_order have full coverage.
+            covered, = np.isclose(hspmap[pix_shift], 4**(max_order - uniq_order)).nonzero()
+            if covered.size == 0:
+                # No pixels at uniq_order are fully covered, we're done.
+                break
+            # Replace the UNIQ pixels that are fully covered.
+            uniq[covered] = 4*(4**uniq_order) + pix_shift[covered]
+
+        # Remove duplicate pixels.
+        uniq = np.unique(uniq)
+
+        # Output to fits.
+        tbl = np.zeros(uniq.size, dtype=[("UNIQ", "i8")])
+        tbl["UNIQ"] = uniq
+
+        order = np.log2(tbl["UNIQ"]//4).astype(np.int32)//2
+        moc_order = np.max(order)
+
+        hdu = fits.BinTableHDU(tbl)
+        hdu.header["PIXTYPE"] = "HEALPIX"
+        hdu.header["ORDERING"] = "NUNIQ"
+        hdu.header["COORDSYS"] = "C"
+        hdu.header["MOCORDER"] = moc_order
+        hdu.header["MOCTOOL"] = "lsst.pipe.tasks.hips.GenerateHipsTask"
+
+        uri = hips_base_path.join("Moc.fits")
+
+        with ResourcePath.temporary_uri(suffix=uri.getExtension()) as temporary_uri:
+            hdu.writeto(temporary_uri.ospath)
+
+            uri.transfer_from(temporary_uri, transfer="copy", overwrite=True)
+
+    def _write_allsky_file(self, hips_base_path, allsky_order):
+        """Write an Allsky.png file.
+
+        Parameters
+        ----------
+        hips_base_path : `lsst.resources.ResourcePath`
+            Resource path to the base of the HiPS directory tree.
+        allsky_order : `int`
+            HEALPix order of the minimum order to make allsky file.
+        """
+        tile_size = self.config.allsky_tilesize
+        n_tiles_per_side = int(np.sqrt(hp.order2npix(allsky_order)))
+
+        allsky_image = None
+
+        allsky_order_uri = hips_base_path.join(f'Norder{allsky_order}', forceDirectory=True)
+        png_uris = list(
+            ResourcePath.findFileResources(
+                candidates=[allsky_order_uri],
+                file_filter=re.compile(r'Npix.*\.png'),
+            )
+        )
+
+        for png_uri in png_uris:
+            matches = re.search("Npix([0-9]+).png", png_uri.geturl())
```

Comment:
It seems like this regex could match the `file_filter` regex above to guarantee that you won't have a situation where `matches` doesn't match. In theory you should have a `$` at the end of this regex and if you use `png_uri.basename()` rather than `geturl` you can make it even more specific.

```python
matches = re.match("Npix([0-9]+).png$", png_uri.basename())
```